core/src/main/scala/cromwell/core/callcaching/CallCachingEligibility.scala

```diff
@@ -1,22 +1,29 @@
 package cromwell.core.callcaching
 
-sealed trait CallCachingEligibility
-case object CallCachingEligible extends CallCachingEligibility
+sealed trait CallCachingEligibility {
+  def isEligible: Boolean
+  def dockerHash: Option[String]
+}
+
+sealed trait CallCachingEligible extends CallCachingEligibility {
+  def isEligible = true
+}
 sealed trait CallCachingIneligible extends CallCachingEligibility {
-  def message: String
+  def isEligible = false
 }
-  
-case class FloatingDockerTagWithHash(hash: String) extends CallCachingIneligible {
-  override val message = s"""You are using a floating docker tag in this task. Cromwell does not consider tasks with floating tags to be eligible for call caching.
-        |If you want this task to be eligible for call caching in the future, use a docker runtime attribute with a digest instead.
-        |This is the exact docker image that was used for this job: $hash
-        |You can replace the docker runtime attribute in your task with the above value to make this task eligible for call caching.""".stripMargin
+
+case object NoDocker extends CallCachingEligible {
+  override def dockerHash: Option[String] = None
+}
+case class DockerWithHash(dockerAttribute: String) extends CallCachingEligible {
+  override def dockerHash: Option[String] = Option(dockerAttribute)
+}
+
+// Currently unused
```

Comment:
Currently it is removable.  I'm not sure why it was added in the first place, so I was inclined to leave if the use case for it is just around the corner.  But I could be convinced otherwise.